src/software/world/ball_state.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <vector>
+
+#include "software/geom/point.h"
+#include "software/util/time/timestamp.h"
+
+class BallState final
+{
+   public:
+    /**
+     * Creates a new ball state with the given position, velocity, and timestamp
+     *
+     * @param position The position of the ball, with coordinates in metres
+     * @param velocity The velocity of the ball, in metres per second
+     * @param timestamp The timestamp at which the ball was observed to be at the
+     * given position and velocity
+     */
+    explicit BallState(Point position, Vector velocity, const Timestamp &timestamp);
+
+    /**
+     * Returns the position of the ball represented by this state
+     *
+     * @return the position of the ball represented by this state
+     */
+    Point position() const;
+
+    /**
+     * Returns the velocity of the ball represented by this state
+     *
+     * @return the velocity of the ball represented by this state
+     */
+    Vector velocity() const;
+
+    /**
+     * Returns the timestamp of the ball represented by this state
+     *
+     * @return the timestamp of the ball represented by this state
+     */
+    Timestamp timestamp() const;
+
+    /**
+     * Defines the equality operator for a BallState. BallStates are equal if their
+     * positions and velocities are the same
+     *
+     * @param other The ball state to compare against for equality
+     * @return True if the other ball state is equal to this ball state, and false
+     * otherwise
+     */
+    bool operator==(const BallState &other) const;
+
+    /**
+     * Defines the inequality operator for a BallState.
+     *
+     * @param other The ball state to compare against for inequality
+     * @return True if the other ball state is not equal to this ball state, and false
+     * otherwise
+     */
+    bool operator!=(const BallState &other) const;
+
+   private:
+    Point pos;
```

Comment:
nit, but could we use full names, especially given that nothing outside this class has to use these names (ie. `position`, `velocity`)? Characters are cheaper then mental cycles :stuck_out_tongue:

Reply:
clion yells at me when I try to do this because of the functions named "position()" and "velocity()"

Reply:
You could suffix the member variables with an underscore, like `position_`

Reply:
yeah, or just let CLion yell at you in this case.

Reply:
no I don't think it will compile if you have a member with the same name as a function.

Reply:
Whoops, missed that one. I'd definitely advocate for just suffixing it with `_` then, rather then abbreviating it.

Reply:
updated